From b1ed6e1741641318c851b8b3241e7f5c6c5990ee Mon Sep 17 00:00:00 2001
From: Paul Cercueil <paul@crapouillou.net>
Date: Wed, 31 Mar 2021 20:26:20 +0100
Subject: [PATCH 3/5] Allow get_param_offset() to retrieve 0th argument

This will be useful later to retrieve storage names in command
arguments.

Signed-off-by: Paul Cercueil <paul@crapouillou.net>
---
 src/mtp_cfg.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/src/mtp_cfg.c b/src/mtp_cfg.c
index 1df7fa1..93ca725 100644
--- a/src/mtp_cfg.c
+++ b/src/mtp_cfg.c
@@ -159,8 +159,7 @@ static int get_param_offset(char * line, int param)
 	offs = 0;
 	offs = get_next_word(line, offs);
 
-	param_cnt = 0;
-	do
+	for (param_cnt = 0; param_cnt < param; param_cnt++)
 	{
 		offs = copy_param(NULL, line, offs);
 
@@ -168,9 +167,7 @@ static int get_param_offset(char * line, int param)
 
 		if(line[offs] == 0 || line[offs] == '#')
 			return -1;
-
-		param_cnt++;
-	}while( param_cnt < param );
+	}
 
 	return offs;
 }
-- 
2.30.2

